src/containers/MainContainer.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { initalize, getRepo } from '../modules/repo';
+import { RootState } from '../modules';
+import MainComponent from '../components/MainComponent';
+import Loading from '../components/Loading';
+
+const MainContainer: React.FC = () => {
+  const [username, setUsername] = useState<string>('');
+  const dispatch = useDispatch();
+  const {
+    repos,
+    loading,
+  } = useSelector(({ repo }: RootState) => repo);
+
+  const handleChangeInput = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setUsername(value);
+  }, [])
+
+  const handleSearch = useCallback((e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    dispatch(getRepo(username));
+  }, [username, dispatch]);
+
+  useEffect(() => {
+    return () => {
+      dispatch(initalize);
+    }
+  }, [dispatch]);
+
+  return (
+    <Loading loading={loading}>
+      {
+        () => (
+          <MainComponent
+            repos={repos}
+            username={username}
+            onChangeInput={handleChangeInput}
+            onSearch={handleSearch}
+          />
+        )
+      }
+    </Loading>
+  );
```

Comment:
바로 component를 넘겨도 정상적으로 작동할 것 같다고 생각하는데 위의 방법으로 작성해주신 이유가 있으신가요??
```suggestion
  return (
    <Loading loading={loading}>
       <MainComponent
         repos={repos}
         username={username}
         onChangeInput={handleChangeInput}
         onSearch={handleSearch}
       />
    </Loading>
  );
```

그리고, `Loading `component가 `MainComponent`를 감싸고 있는 형태로 작성해주신 이유가 있을까요?? 
loading state에 따라서 `Loading `component를 보여줄지 안 보여줄지 결정해도 될것 같다고 생각해서요!

Reply:
저도 궁금합니다.

Reply:
이 부분에 관해서 다른 방법으로 LoadingComponent 상속 받아서 하는거나
customhooks 만들어 봐도 좋을거 같아요 !

Reply:
저도 처음에 @YukJiSoo님 말씀대로 `loading` 값에 따라 조건분기해서 렌더링을 해줬었는데 그냥 프로젝트 하다가 `Loading`컴포넌트는 렌더속성값으로 바꿔주고 싶어져서 저렇게 했습니다ㅎㅎ 사실 렌더링비용 측면에선 @YukJiSoo님이 말씀해주신 방법대로 하는게 더 좋을듯 합니다